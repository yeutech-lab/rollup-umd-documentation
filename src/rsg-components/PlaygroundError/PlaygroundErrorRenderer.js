import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

const styles = ({ fontFamily, fontSize, color }) => ({
	root: {
		margin: 0,
		lineHeight: 1.2,
		fontSize: fontSize.small,
		fontFamily: fontFamily.monospace,
		color: color.error,
		whiteSpace: 'pre',
	},
});

export function PlaygroundErrorRenderer({ classes, message }) {
	return <pre className={classes.root}>{message}</pre>;
}

PlaygroundErrorRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	message: PropTypes.string.isRequired,
};

export default Styled(styles)(PlaygroundErrorRenderer);
