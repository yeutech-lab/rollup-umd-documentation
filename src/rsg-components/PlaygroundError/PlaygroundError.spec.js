import React from 'react';
import { PlaygroundErrorRenderer } from './PlaygroundErrorRenderer';

it('renderer should render message', () => {
	const message = 'Hello *world*!';
	const actual = shallow(<PlaygroundErrorRenderer classes={{}} message={message} />);

	expect(actual).toMatchSnapshot();
});
