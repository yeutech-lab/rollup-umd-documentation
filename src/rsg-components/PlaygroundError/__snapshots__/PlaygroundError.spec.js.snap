// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render message 1`] = `
<pre>
  Hello *world*!
</pre>
`;
